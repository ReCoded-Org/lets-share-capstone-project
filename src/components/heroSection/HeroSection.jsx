import * as React from "react";
import Image from "next/image";
import { IoIosHelpBuoy } from "react-icons/io";

const HeroSection = () => {
    return (
        <div>
            <div
                id='firstSection'
                className='mt-[70px] grid grid-cols-1 items-center px-5 md:mx-20 md:grid-cols-2 md:gap-20'
            >
                <div className='md:order-1'>
                    <Image
                        src='/../public/images/undraw.png'
                        width={850}
                        height={700}
                        alt='/'
                    />
                </div>
                <div className='my-[30px] flex flex-col items-center justify-center text-center'>
                    <p className='mb-8 py-5 text-2xl md:text-3xl'>
                        Lorem ipsum dolor sit amet, consectetur hello adipiscing
                        elit, sed do eiusmod tempor incididunt ut labore et
                        dolore magna aliqua. Ut enim ad minim veniam, quis
                        nostrud exercitation ullamco laboris nisi ut aliquip ex
                        ea commodo consequat.
                    </p>
                    <button
                        type='submit'
                        className='rounded-full bg-primary py-3 px-6 text-3xl font-bold  text-[white]'
                    >
                        donate Now
                    </button>
                </div>
            </div>
            <hr className='my-10 mx-20 border-primary' />
            <div id='causes' className='my-10'>
                <h2 className='text-center text-4xl font-black text-fontColor'>
                    Causes
                </h2>
                <div className='grid grid-cols-1 items-center md:grid-cols-2 lg:grid-cols-3 '>
                    <div className='mx-10 my-10 flex flex-col items-center gap-5 px-5'>
                        <IoIosHelpBuoy size={130} color='#F07167' />
                        <h3 className='text-2xl font-bold text-fontColor'>
                            title
                        </h3>
                        <p className='text-center text-lg md:text-xl'>
                            Lorem ipsum dolor, sit amet consectetur adipisicing
                            elit. Necessitatibus deleniti minima voluptatem
                            autem sequi, ut non. Qui harum quisquam asperiores
                            impedit, quas velit consectetur ducimus adipisci et
                            fugit placeat blanditiis.
                        </p>
                    </div>
                    <div className='mx-10 my-10 flex flex-col items-center gap-5 px-5'>
                        <IoIosHelpBuoy size={130} color='#F07167' />
                        <h3 className='text-2xl font-bold text-fontColor'>
                            title
                        </h3>
                        <p className='text-center text-lg'>
                            Lorem ipsum dolor, sit amet consectetur adipisicing
                            elit. Necessitatibus deleniti minima voluptatem
                            autem sequi, ut non. Qui harum quisquam asperiores
                            impedit, quas velit consectetur ducimus adipisci et
                            fugit placeat blanditiis.
                        </p>
                    </div>
                    <div className='mx-10 my-10 flex flex-col items-center gap-5 px-5'>
                        <IoIosHelpBuoy size={130} color='#F07167' />
                        <h3 className='text-2xl font-bold text-fontColor'>
                            title
                        </h3>
                        <p className='text-center text-lg'>
                            Lorem ipsum dolor, sit amet consectetur adipisicing
                            elit. Necessitatibus deleniti minima voluptatem
                            autem sequi, ut non. Qui harum quisquam asperiores
                            impedit, quas velit consectetur ducimus adipisci et
                            fugit placeat blanditiis.
                        </p>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default HeroSection;
