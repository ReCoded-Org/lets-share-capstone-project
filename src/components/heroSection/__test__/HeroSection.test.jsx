import renderer from "react-test-renderer";
import HeroSection from "../HeroSection";
import * as nextRouter from "next/router";
it("renders correctly", () => {
    const tree = renderer.create(<HeroSection />).toJSON();
    expect(tree).toMatchSnapshot();
});
function tryRoute(packacge) {
    packacge.useRouter = jest.fn();
    packacge.useRouter.mockImplementation(() => ({ route: "/" }));
}
tryRoute(nextRouter);
