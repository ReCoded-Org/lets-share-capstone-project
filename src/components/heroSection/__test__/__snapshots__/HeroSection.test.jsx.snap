// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="mt-[70px] grid grid-cols-1 items-center px-5 md:mx-20 md:grid-cols-2 md:gap-20"
    id="firstSection"
  >
    <div
      className="md:order-1"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27850%27%20height=%27700%27/%3e"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="/"
          data-nimg="intrinsic"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="/"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2F..%2Fpublic%2Fimages%2Fundraw.png&w=1920&q=75"
            srcSet="/_next/image?url=%2F..%2Fpublic%2Fimages%2Fundraw.png&w=1080&q=75 1x, /_next/image?url=%2F..%2Fpublic%2Fimages%2Fundraw.png&w=1920&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
    </div>
    <div
      className="my-[30px] flex flex-col items-center justify-center text-center"
    >
      <p
        className="mb-8 py-5 text-2xl md:text-3xl"
      >
        heroMain
      </p>
      <button
        className="rounded-full bg-primary py-3 px-6 text-3xl font-bold text-[white]"
        type="submit"
      >
        donate now
      </button>
    </div>
  </div>
  <hr
    className="my-10 mx-20 border-primary"
  />
  <div
    className="my-10"
    id="causes"
  >
    <h2
      className="text-center text-4xl font-black text-fontColor"
    >
      causes
    </h2>
    <div
      className="grid grid-cols-1 items-center md:grid-cols-2 lg:grid-cols-3 "
    >
      <div
        className="mx-10 my-10 flex flex-col items-center gap-5 px-5"
      >
        <svg
          color="#F07167"
          fill="currentColor"
          height={130}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#F07167",
            }
          }
          viewBox="0 0 512 512"
          width={130}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm54.6 36.4c27.1 8.6 52 23.6 72.7 44.3 20.7 20.7 35.7 45.6 44.3 72.7l-88.8 6c-8.2-14-19.9-25.7-33.9-34l5.7-89zm-181.9 44.3c20.7-20.7 45.5-35.7 72.7-44.3l5.7 89c-13.9 8.3-25.6 20-33.9 33.9l-88.9-5.9c8.7-27.1 23.7-52 44.4-72.7zm72.7 298.9c-27.1-8.6-52-23.6-72.7-44.3-20.7-20.7-35.7-45.6-44.3-72.7l89-5.7c8.2 13.9 19.9 25.5 33.8 33.8l-5.8 88.9zM256 324c-37.5 0-68-30.5-68-68s30.5-68 68-68 68 30.5 68 68-30.5 68-68 68zm127.3 59.3c-20.7 20.7-45.6 35.7-72.7 44.3l-5.9-88.9c14.1-8.3 25.8-20.1 34.1-34.2l88.8 6c-8.6 27.2-23.6 52.1-44.3 72.8z"
          />
        </svg>
        <h3
          className="text-2xl font-bold text-fontColor"
        >
          causeTitle1
        </h3>
        <p
          className="text-center text-lg md:text-xl"
        >
          causeText1
        </p>
      </div>
      <div
        className="mx-10 my-10 flex flex-col items-center gap-5 px-5"
      >
        <svg
          color="#F07167"
          fill="currentColor"
          height={130}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#F07167",
            }
          }
          viewBox="0 0 512 512"
          width={130}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm54.6 36.4c27.1 8.6 52 23.6 72.7 44.3 20.7 20.7 35.7 45.6 44.3 72.7l-88.8 6c-8.2-14-19.9-25.7-33.9-34l5.7-89zm-181.9 44.3c20.7-20.7 45.5-35.7 72.7-44.3l5.7 89c-13.9 8.3-25.6 20-33.9 33.9l-88.9-5.9c8.7-27.1 23.7-52 44.4-72.7zm72.7 298.9c-27.1-8.6-52-23.6-72.7-44.3-20.7-20.7-35.7-45.6-44.3-72.7l89-5.7c8.2 13.9 19.9 25.5 33.8 33.8l-5.8 88.9zM256 324c-37.5 0-68-30.5-68-68s30.5-68 68-68 68 30.5 68 68-30.5 68-68 68zm127.3 59.3c-20.7 20.7-45.6 35.7-72.7 44.3l-5.9-88.9c14.1-8.3 25.8-20.1 34.1-34.2l88.8 6c-8.6 27.2-23.6 52.1-44.3 72.8z"
          />
        </svg>
        <h3
          className="text-2xl font-bold text-fontColor"
        >
          causeTitle2
        </h3>
        <p
          className="text-center text-lg"
        >
          causeText2
        </p>
      </div>
      <div
        className="mx-10 my-10 flex flex-col items-center gap-5 px-5"
      >
        <svg
          color="#F07167"
          fill="currentColor"
          height={130}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#F07167",
            }
          }
          viewBox="0 0 512 512"
          width={130}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm54.6 36.4c27.1 8.6 52 23.6 72.7 44.3 20.7 20.7 35.7 45.6 44.3 72.7l-88.8 6c-8.2-14-19.9-25.7-33.9-34l5.7-89zm-181.9 44.3c20.7-20.7 45.5-35.7 72.7-44.3l5.7 89c-13.9 8.3-25.6 20-33.9 33.9l-88.9-5.9c8.7-27.1 23.7-52 44.4-72.7zm72.7 298.9c-27.1-8.6-52-23.6-72.7-44.3-20.7-20.7-35.7-45.6-44.3-72.7l89-5.7c8.2 13.9 19.9 25.5 33.8 33.8l-5.8 88.9zM256 324c-37.5 0-68-30.5-68-68s30.5-68 68-68 68 30.5 68 68-30.5 68-68 68zm127.3 59.3c-20.7 20.7-45.6 35.7-72.7 44.3l-5.9-88.9c14.1-8.3 25.8-20.1 34.1-34.2l88.8 6c-8.6 27.2-23.6 52.1-44.3 72.8z"
          />
        </svg>
        <h3
          className="text-2xl font-bold text-fontColor"
        >
          causeTitle3
        </h3>
        <p
          className="text-center text-lg"
        >
          causeText3
        </p>
      </div>
    </div>
  </div>
</div>
`;
